src/addrdb.cpp

```diff
@@ -197,9 +197,12 @@ std::optional<bilingual_str> LoadAddrman(const std::vector<bool>& asmap, const A
         LogPrintf("Creating peers.dat because the file was not found (%s)\n", fs::quoted(fs::PathToString(path_addr)));
         DumpPeerAddresses(args, *addrman);
     } catch (const InvalidAddrManVersionError&) {
+        if (!RenameOver(path_addr, (fs::path)path_addr + ".bak")) {
+            throw std::exception();
```

Comment:
```C++
throw std::runtime_error{"Failed to rename invalid peer.dat file. Please move or delete it try again."};
```

Reply:
Doesn't this error get caught by the catch below (which catches exception)?

Reply:
Ah yeah didn't realize this was caught later, I thought you just hard-crashed. Either way, you do want it to be caught and handled below, but the wording here is a bit redundant. Maybe just "Failed to move invalid peer.dat file aside."?

Reply:
Actually, it looks like throwing here bypasses the later catch.

Reply:
Maybe I should refactor the other comment from Luke to make things cleaner.